staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -200,8 +200,7 @@ func (d *DiscoveryClient) serverResources() ([]*metav1.APIResourceList, error) {
 		for _, version := range apiGroup.Versions {
 			gv := schema.GroupVersion{Group: apiGroup.Name, Version: version.Version}
 			resources, err := d.ServerResourcesForGroupVersion(version.GroupVersion)
-			if err != nil {
-				// TODO: maybe restrict this to NotFound errors
+			if err != nil && errors.IsNotFound(err) {
```

Comment:
Simply changing to this is not correct.

What if `err` is not `nil`, but `errors.IsNotFound(err)` is false?